### PR TITLE
Update Azure container registry

### PR DESCRIPTION
**Pull request by milismsft**
Update API version and inner classes.
Update implementation and add support for managed container registry and container registry webhooks.
Update tests, samples and recording.

### CODE REVIEW

File: src/ResourceManagement/ContainerRegistry/RegistriesImpl.cs [hunk 1]
```diff
@@ -5,117 +5,168 @@ namespace Microsoft.Azure.Management.ContainerRegistry.Fluent
...
+            return new RegistryCredentialsImpl(Extensions.Synchronize(() => this.Inner.RegenerateCredentialAsync(resourceGroupName, registryName, accessKeyType.ToPasswordName())));
         }
 
-        protected async override Task<IPage<Models.RegistryInner>> ListInnerByGroupAsync(string groupName, CancellationToken cancellationToken)
+        ///GENMHASH:5B0E9060E6A010D9AEC0DEE8C817853F:81A8B8E3661A768A3015CC178EFC3516
+        public async Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.IRegistryCredentials> RegenerateCredentialAsync(string resourceGroupName, string registryName, AccessKeyType accessKeyType, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return await this.Inner.ListByResourceGroupAsync(groupName, cancellationToken);
+            return new RegistryCredentialsImpl(await this.Inner.RegenerateCredentialAsync(resourceGroupName, registryName, accessKeyType.ToPasswordName()));
         }
```
Comment (hovsepm) [inline src/ResourceManagement/ContainerRegistry/RegistriesImpl.cs]: please pass cancellation token to async call

File: src/ResourceManagement/ContainerRegistry/RegistryImpl.cs [hunk 2]
```diff
@@ -6,215 +6,323 @@ namespace Microsoft.Azure.Management.ContainerRegistry.Fluent
...
             if (IsInCreateMode)
             {
-                var storageAccountParameters = await this.HandleStorageSettingsAsync(cancellationToken);
-                createParameters.StorageAccount = storageAccountParameters;
-                createParameters.Location = this.RegionName.ToLower();
-                createParameters.Tags = this.Inner.Tags;
-                var registryInner = await this.Manager.Inner.Registries.CreateAsync(this.ResourceGroupName, this.Name, createParameters);
+                if (this.creatableStorageAccountKey != null)
+                {
+                    IStorageAccount storageAccount = this.CreatedResource(this.creatableStorageAccountKey) as IStorageAccount;
```
Comment (hovsepm) [inline src/ResourceManagement/ContainerRegistry/RegistryImpl.cs]: >CreatedResource [](start = 58, length = 15)

Are you making sync call in async method?

Comment (hovsepm) [inline src/ResourceManagement/ContainerRegistry/RegistryImpl.cs]: >as IStorageAccount [](start = 107, length = 18)

Please use direct cast.

File: src/ResourceManagement/ContainerRegistry/WebhookOperationsImpl.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                return new List<IWebhook>().AsReadOnly();
+            }
+        }
+
+        ///GENMHASH:7F5BEBF638B801886F5E13E6CCFF6A4E:653839A0F5B936DE3781953080508EA4
+        public async Task<Microsoft.Azure.Management.ResourceManager.Fluent.Core.IPagedCollection<IWebhook>> ListAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (this.containerRegistry != null)
+            {
+                return await this.webhooksClient.ListAsync(this.containerRegistry.ResourceGroupName, this.containerRegistry.Name);
```
Comment (hovsepm) [inline src/ResourceManagement/ContainerRegistry/WebhookOperationsImpl.cs]: >Name [](start = 124, length = 4)

please pass in cancellationToken

Comment (milismsft) [reply src/ResourceManagement/ContainerRegistry/WebhookOperationsImpl.cs]: done

File: src/ResourceManagement/ContainerRegistry/RegistryImpl.cs (hunk 2 above)
Comment (milismsft) [reply src/ResourceManagement/ContainerRegistry/RegistryImpl.cs]: nope

Comment (milismsft) [reply src/ResourceManagement/ContainerRegistry/RegistryImpl.cs]: reworked the code so it does not need a cast.

File: src/ResourceManagement/ContainerRegistry/RegistriesImpl.cs (hunk 1 above)
Comment (milismsft) [reply src/ResourceManagement/ContainerRegistry/RegistriesImpl.cs]: done

**Comment by hovsepm** [comment]
@azuresdkci retest this please

**Pull request merged**